openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/PipelineServiceClientFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.service.clients.pipeline;
+
+import java.lang.reflect.InvocationTargetException;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.api.configuration.pipelineServiceClient.PipelineServiceClientConfiguration;
+import org.openmetadata.sdk.PipelineServiceClient;
+
+@Slf4j
+public class PipelineServiceClientFactory {
+
+  @Getter private static PipelineServiceClient pipelineServiceClient;
+
+  public static PipelineServiceClient createPipelineServiceClient(PipelineServiceClientConfiguration config) {
+    if (pipelineServiceClient != null) {
+      return pipelineServiceClient;
+    }
+
+    String pipelineServiceClientClass = config.getClassName();
+    LOG.info("Registering PipelineServiceClient: {}", pipelineServiceClientClass);
+
+    try {
+      pipelineServiceClient =
+          Class.forName(pipelineServiceClientClass)
+              .asSubclass(PipelineServiceClient.class)
+              .getConstructor(PipelineServiceClientConfiguration.class)
+              .newInstance(config);
+      return pipelineServiceClient;
+    } catch (ClassNotFoundException
+        | NoSuchMethodException
+        | InvocationTargetException
+        | InstantiationException
+        | IllegalAccessException e) {
+      LOG.error("Error trying to load PipelineServiceClient {}: {}", pipelineServiceClientClass, e.getMessage());
```

Comment:
Not sure if we want this to blow up harder. The train of thought here was in cases where we just want the server without any pipeline service client integration

Reply:
I think we should throw an exception here and stop the server from starting. Otherwise, as soon the server checks the status of the pipeline service, it will throw a `NullPointerException`. We should have a `NoopPipelineService` implementation if someone just wants to use the OM server and ingest everything manually.